File: CodingPractice-master/leetcode/algorithms/cpp/0509_FibonacciNumber.cpp
/*
* @Author: Yan_Daojiang
* @Date:   2019-05-09 11:22:30
* @Last Modified by:   Yan_Daojiang
* @Last Modified time: 2019-05-09 11:22:45
*/
class Solution {
public:
    int fib(int N) {
        int dp[31]={0,1,1};
        for(int i=3;i<=N;i++)
            dp[i]=dp[i-1]+dp[i-2];
        return dp[N];
    }
};

File: CodingPractice-master/leetcode/algorithms/python/0509_FibonacciNumber.py
# -*- coding: utf-8 -*-
# @Author: Yan_Daojiang
# @Date:   2019-05-09 11:21:29
# @Last Modified by:   Yan_Daojiang
# @Last Modified time: 2019-05-09 11:21:42
class Solution:
    def fib(self, N: int) -> int:
        dp=[0,1,1]
        for i in range(3,N+1):
            dp.append(dp[i-1]+dp[i-2])
        return dp[N]